_posts/2024-11-13-boosting-k-nn-exact-search.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title: "Boosting k-NN exact search performance"
+authors:
+   - ryanbogan
+   - jmazane
+   - vamshin
+   - kolchfa
+date: 2024-11-19
+categories: 
+    - technical-posts
+has_science_table: true
+meta_keywords: k-NN search performance, SIMD in OpenSearch, script score queries, performance optimization, machine learning, exact K-NN
+meta_description: Boost exact k-NN search performance in OpenSearch using SIMD optimizations and script_score queries. Discover real-world performance gains for efficient vector similarity searches in machine learning applications.
+---
+
+Exact k-nearest neighbor (k-NN) search in OpenSearch allows you to define custom scoring functions to retrieve documents based on their proximity to a query vector. This method provides highly accurate search results, making it ideal when you need precise, deterministic matches.
+
+Using OpenSearch's `script_score` queries, you can perform exact k-NN searches to find the closest neighbors to a query vector. This query type allows you to create complex scoring functions that account for factors like document attributes, user preferences, or external data. 
```

Comment:
Add a link to here: https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/